File: src/main/java/org/domain/erp/entity/Person.java
package org.domain.erp.entity;

import java.math.BigDecimal;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.IdClass;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

import org.domain.crud.entity.CompanyIdPK;

@IdClass(CompanyIdPK.class)
@Entity
@Table(name = "person")
public class Person implements java.io.Serializable {
	/**
	 *
	 */
	private static final long serialVersionUID = -5971740408816408928L;
	@Id private Integer company;
	@Id
	@SequenceGenerator(name="person_id_seq", sequenceName="person_id_seq", allocationSize=1)
	@GeneratedValue(strategy=GenerationType.SEQUENCE, generator="person_id_seq")
	@Column(columnDefinition="serial")
	private Integer id;
	@Column(name = "name", unique = true, nullable = false, length = 100)
	private String name;
	@Column(name = "cnpj_cpf", unique = true, length = 18)
	private String cnpjCpf;
	@Column(name = "ie_rg", unique = true, length = 12)
	private String ieRg;
   @Column(name = "suframa", columnDefinition = "character varying(9)")
   private String suframa;
   @Column(name = "im", columnDefinition = "character varying(12)")
   private String im;
	@Column(name = "cnae")//][ForeignKey("IbgeCnae")
	private Integer cnae;
	// 1=Simples Nacional; 2=Simples Nacional, excesso sublimite de receita bruta; 3=Regime Normal. (v2.0).
	@Column(name = "crt")//][FilterUIHint("", "", "defaultValue", "1", "options", "1 - Simples Nacional,2 - Simples Nacional (excesso sublimite de receita bruta),3 - Regime Normal")
	private Integer crt;
	@Column(name = "zip", length = 9, columnDefinition="bpchar(9)")
	private String zip;
	@Column(name = "country")//[ForeignKey("BacenCountry")
	private Integer country;
	private Integer uf;
	private Integer city;
	@Column(name = "district", length = 64)
	private String district;
	@Column(name = "address", length = 100)
	private String address;
   @Column(name = "address_number", columnDefinition = "character varying(16)")
   private String addressNumber;
   @Column(name = "complement", columnDefinition = "character varying(16)")
   private String complement;
	@Column(name = "email", length = 100)
	private String email;
	@Column(name = "site", length = 100)
	private String site;
	@Column(name = "phone", length = 16)
	private String phone;
	@Column(name = "fax", length = 16)
	private String fax;
	@Column(name = "credit", precision = 9, scale = 3)
	private BigDecimal credit;
	@Column(name = "additional_data")
	private String additionalData;

	public Person() {
	}


	public Integer getCompany() {
		return company;
	}

	public void setCompany(Integer company) {
		this.company = company;
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getCnpjCpf() {
		return cnpjCpf;
	}

	public void setCnpjCpf(String cnpjCpf) {
		this.cnpjCpf = cnpjCpf;
	}

	public String getIeRg() {
		return ieRg;
	}

	public void setIeRg(String ieRg) {
		this.ieRg = ieRg;
	}

	public String getSuframa() {
		return suframa;
	}

	public void setSuframa(String suframa) {
		this.suframa = suframa;
	}

	public String getIm() {
		return im;
	}

	public void setIm(String im) {
		this.im = im;
	}

	public Integer getCnae() {
		return cnae;
	}

	public void setCnae(Integer cnae) {
		this.cnae = cnae;
	}

	public Integer getCrt() {
		return crt;
	}

	public void setCrt(Integer crt) {
		this.crt = crt;
	}

	public String getZip() {
		return zip;
	}

	public void setZip(String zip) {
		this.zip = zip;
	}

	public Integer getCountry() {
		return country;
	}

	public void setCountry(Integer country) {
		this.country = country;
	}

	public Integer getUf() {
		return uf;
	}

	public void setUf(Integer uf) {
		this.uf = uf;
	}

	public Integer getCity() {
		return city;
	}

	public void setCity(Integer city) {
		this.city = city;
	}

	public String getDistrict() {
		return district;
	}

	public void setDistrict(String district) {
		this.district = district;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public String getAddressNumber() {
		return addressNumber;
	}

	public void setAddressNumber(String addressNumber) {
		this.addressNumber = addressNumber;
	}

	public String getComplement() {
		return complement;
	}

	public void setComplement(String complement) {
		this.complement = complement;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getSite() {
		return site;
	}

	public void setSite(String site) {
		this.site = site;
	}

	public String getPhone() {
		return phone;
	}

	public void setPhone(String phone) {
		this.phone = phone;
	}

	public String getFax() {
		return fax;
	}

	public void setFax(String fax) {
		this.fax = fax;
	}

	public BigDecimal getCredit() {
		return credit;
	}

	public void setCredit(BigDecimal credit) {
		this.credit = credit;
	}

	public String getAdditionalData() {
		return additionalData;
	}

	public void setAdditionalData(String additionalData) {
		this.additionalData = additionalData;
	}

}

File: src/main/dotnet/erp/Entity/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AspNetCoreWebApi.Entity
{
    [Table("person")]
    public partial class Person
    {
        public Person()
        {
/*
            Request = new HashSet<Request>();
            RequestFreight = new HashSet<RequestFreight>();
            RequestNfe = new HashSet<RequestNfe>();
 */
        }

        [Column("crud_group_owner")]
        public int? CrudGroupOwner { get; set; }
        [Key][Column("id")][DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        [Column("name", TypeName = "character varying(100)")]
        public string Name { get; set; }
        [Column("fantasy", TypeName = "character varying(100)")]
        public string Fantasy { get; set; }
        [Column("cnpj_cpf", TypeName = "character varying(18)")]
        public string CnpjCpf { get; set; }
        [Column("ie_rg", TypeName = "character varying(12)")]
        public string IeRg { get; set; }
        [Column("suframa", TypeName = "character varying(9)")]
        public string Suframa { get; set; }
        [Column("im", TypeName = "character varying(12)")]
        public string Im { get; set; }
		[Column("cnae")][ForeignKey("IbgeCnae")]
        public int? Cnae { get; set; }
		// 1=Simples Nacional; 2=Simples Nacional, excesso sublimite de receita bruta; 3=Regime Normal. (v2.0).
		[Column("crt")][FilterUIHint("", "", "defaultValue", "1", "options", "1 - Simples Nacional,2 - Simples Nacional (excesso sublimite de receita bruta),3 - Regime Normal")]
        public int? Crt { get; set; }
        [Column("zip", TypeName = "character(9)")]
        public string Zip { get; set; }
		[Column("country")][ForeignKey("BacenCountry")]
        public int? Country { get; set; }
		[Column("uf")][ForeignKey("IbgeUf")]
        public int? Uf { get; set; }
		[Column("city")][ForeignKey("IbgeCity")]
        public int? City { get; set; }
        [Column("district", TypeName = "character varying(64)")]
        public string District { get; set; }
        [Column("address", TypeName = "character varying(100)")]
        public string Address { get; set; }
        [Column("address_number", TypeName = "character varying(16)")]
        public string AddressNumber { get; set; }
        [Column("complement", TypeName = "character varying(16)")]
        public string Complement { get; set; }
        [Column("email", TypeName = "character varying(100)")]
        public string Email { get; set; }
        [Column("site", TypeName = "character varying(100)")]
        public string Site { get; set; }
        [Column("phone", TypeName = "character varying(16)")]
        public string Phone { get; set; }
        [Column("fax", TypeName = "character varying(16)")]
        public string Fax { get; set; }
		[Column("credit", TypeName = "numeric(9,3)")][FilterUIHint("", "", "defaultValue", "0.000")]
        public decimal? Credit { get; set; }
        [Column("additional_data", TypeName = "character varying(255)")]
        public string AdditionalData { get; set; }
/*
        [ForeignKey("City")]
        [InverseProperty("Person")]
        public IbgeCity CityNavigation { get; set; }
        [ForeignKey("Cnae")]
        [InverseProperty("Person")]
        public IbgeCnae CnaeNavigation { get; set; }
        [ForeignKey("Country")]
        [InverseProperty("Person")]
        public BacenCountry CountryNavigation { get; set; }
        [ForeignKey("Uf")]
        [InverseProperty("Person")]
        public IbgeUf UfNavigation { get; set; }
        [InverseProperty("Person")]
        public Employed Employed { get; set; }
        [InverseProperty("PersonNavigation")]
        public ICollection<Request> Request { get; set; }
        [InverseProperty("PersonNavigation")]
        public ICollection<RequestFreight> RequestFreight { get; set; }
        [InverseProperty("PersonNavigation")]
        public ICollection<RequestNfe> RequestNfe { get; set; }
*/
    }
}
